x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/response/ErrorResponseEntity.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.response;
+
+import org.elasticsearch.xcontent.XContentFactory;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.inference.external.http.HttpResult;
+import org.elasticsearch.xpack.inference.external.http.retry.ErrorMessage;
+
+import java.util.Map;
+
+/**
+ * A pattern is emerging in how external providers provide error responses.
+ *
+ * At a minimum, these return:
+ * {
+ *     "error: {
+ *         "message": "(error message)"
+ *     }
+ * }
+ *
+ * Others may return additional information such as error codes specific to the service.
+ *
+ * This currently covers error handling for Azure AI Studio, however this pattern
+ * can be used to simplify and refactor handling for Azure OpenAI and OpenAI responses.
+ */
+public class ErrorResponseEntity implements ErrorMessage {
```

Comment:
How about we name this to indicate it's for azure/openai?

Reply:
Makes sense... I was thinking about it as it seems that if we start to see more providers using the same / similar responses (and error responses) as OpenAI does then this would be a bit more generic than just those two... however, until we _do_ start to see that / integrate with others that use the same pattern - I think it makes sense to name this (and the response handler base) appropriately as you mention (and then change in the future if it makes sense). Thanks!